fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/provider/FabricLanguageProvider.java

```diff
@@ -207,7 +207,11 @@ default void add(RegistryEntry<EntityAttribute> entityAttribute, String value) {
 		 * @param value    The value of the entry.
 		 */
 		default void add(StatType<?> statType, String value) {
-			add("stat_type." + Registries.STAT_TYPE.getId(statType).toString().replace(':', '.'), value);
+			if (statType.getName() instanceof TranslatableTextContent translatableTextContent) {
+				add(translatableTextContent.getKey(), value);
+			} else { // fallback to Util.createTranslationKey formatting (should it error instead?)
```

Comment:
Error? not sure how the translation would be used if the name is not TTC.

Reply:
I just threw in a fallback because you never know, some person might try and have like some string literal concatenated with the translation. But I can always make it an error.